### PR TITLE
Fix Azure OpenAI setup problems

### PR DESCRIPTION
**Pull request by jayden5744**
### Background

- See this issue: https://github.com/Significant-Gravitas/Auto-GPT/issues/4838

### Changes

- Setting azure config values to openai
- Recover the missing 'get_azure_deployment_id_for_model' function from that commit(#4803)

### Documentation

x

### Test Plan

CI

### PR Quality Checklist
- [x] My pull request is atomic and focuses on a single change.
- [ ] I have thoroughly tested my changes with multiple different prompts.
- [x] I have considered potential risks and mitigations for my changes.
- [ ] I have documented my changes clearly and comprehensively.
- [x] I have not snuck in any "extra" small tweaks changes. 
- [x] I have run the following commands against my code to ensure it passes our linters:
    ```shell
    black .
    isort .
    mypy
    autoflake --remove-all-unused-imports --recursive --ignore-init-module-imports --ignore-pass-after-docstring autogpt tests --in-place
    ```

### CODE REVIEW

**Comment by lc0rp** [comment]
@jayden5744, thanks for the PR. Please write some tests!

File: autogpt/config/config.py [hunk 1]
```diff
@@ -44,11 +45,13 @@ class Config(SystemSettings):
     openai_organization: Optional[str] = None
     temperature: float
     use_azure: bool
+    azure_config_file: str
```
Comment (Pwuts) [inline autogpt/config/config.py]:
```suggestion
    azure_config_file: Optional[str] = None
```

File: autogpt/config/config.py [hunk 2]
```diff
@@ -44,11 +45,13 @@ class Config(SystemSettings):
...
     execute_local_commands: bool
     restrict_to_workspace: bool
     openai_api_type: Optional[str] = None
     openai_api_base: Optional[str] = None
     openai_api_version: Optional[str] = None
+    azure_model_to_deployment_id_map: Optional[Dict[str, str]] = None
```
Comment (Pwuts) [inline autogpt/config/config.py]: Please add this directly under `azure_config_file`

File: autogpt/config/config.py [hunk 3]
```diff
@@ -310,6 +322,33 @@ def check_openai_api_key(config: Config) -> None:
...
+    elif model == config.smart_llm_model:
+        return config.azure_model_to_deployment_id_map[
+            "smart_llm_model_deployment_id"
+        ]  # type: ignore
+    elif model == "text-embedding-ada-002":
+        return config.azure_model_to_deployment_id_map[
+            "embedding_model_deployment_id"
+        ]  # type: ignore
+    else:
+        return ""
```
Comment (Pwuts) [inline autogpt/config/config.py]: Maybe return `None` if no matching deployment ID is available?

**Comment by collijk** [comment]
@jayden5744 Overall great work.  Please see https://github.com/jayden5744/Auto-GPT/pull/1/files for suggested changes.

**Review by collijk — APPROVED** [review]

**Comment by jayden5744** [comment]
@collijk 
My existing code excludes the code below, 
```python
# autogpt/config/config.py
if config_dict["use_azure"]: # line 294 
      azure_config = cls.load_azure_config(config_dict["azure_config_file"])
      config_dict.update(azure_config)
     
     # Add the three lines of code below
      openai.api_type = azure_config["openai_api_type"]
      openai.api_base = azure_config["openai_api_base"]
      openai.api_version = azure_config["openai_api_version"]
``` 
which causes the error openai.error.AuthenticationError: Incorrect API key provided to occur when calling the openai api without changing to the azure settings. 
I need to fix this again.

**Comment by collijk** [comment]
We specifically don't want to do 

```
     # Add the three lines of code below
      openai.api_type = azure_config["openai_api_type"]
      openai.api_base = azure_config["openai_api_base"]
      openai.api_version = azure_config["openai_api_version"]
  ```
  
  as this reaches into module namespace and has global impacts.  These should be passed as arguments to the API call (as they are now).  I'm setting up some testing on azure tomorrow (we unfortunately don't have access as part of our CI) and will try to reproduce your issue.

**Pull request merged**